src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestService.java

```diff
@@ -26,4 +26,17 @@
    * @throws ServiceException if file is not valid, error reading file, if records are empty
    */
   List<ByteArrayInputStream> harvest(String url) throws ServiceException;
+
+
+  /**
+   * Harvest the given endpoint {@link String} to a list of byte[], one string per file in the zip
+   *
+   * @param endpoint    for zip file containing one or more records
+   * @param setSpec     record specification
+   * @param prefix      record prefix
+   * @return List of byte[]
+   * @throws ServiceException if file is not valid, error reading file, if records are empty
+   */
+  List<ByteArrayInputStream> harvest(String endpoint, String setSpec, String prefix) throws ServiceException;
```

Comment:
I'm actually ok either way in this case. You could see these methods as similar to each other, as in, they all do the same thing in the end, but each performs with a different approach. It also shows different type of supports (the input parameters I mean) when it comes to harvesting.
However, I also see the benefits of being explicit with the type of harvesting in the methods names. It helps us distinguish each other and it makes things easier when we're looking for a specific approach.
But in the end, it's just a matter on what everyone agrees :)

Reply:
Actually they do totally different things.
Furthermore one has `harvest(String endpoint, String setSpec, String prefix)` and the other one `harvest(String url)`..
An endpoint is also a url..
Even the javadoc is not clear in fact, one says `Harvest the given URL` and the other one `Harvest the given endpoint`.
What is this endpoint, what are those parameters for this endpoint? Is this new method the same with the url one but with added parameters? I can go on and on here..
And now that I see the javadoc better, it's even saying `@param endpoint    for zip file containing one or more records`

It's definitely not clear if we give this contract to someone who is not the implementer of this code(and none of us since we know what we are doing here).
So for sure the javadoc has to be way more descriptive of what's going on here but in my opinion the method call should also be named differently.

For example the following is way more clear, I don't even need to read the javadoc:
```
harvestZipMultipartFile(MultipartFile file)
harvestZipUrl(String url)
harvestOaipmhEndpoint(String endpoint, String setSpec, String prefix)
```

Reply:
Thanks for pointing this out! I actually thought that each method were similar.
I agree with you then, both the method names and javadoc need improvement and need to be more specific